conans/test/functional/command/test_package_test.py

```diff
@@ -139,7 +139,7 @@ class Pkg(ConanFile):
                 requires = "dep/1.1"
                 def build(self):
                     info = self.deps_cpp_info["dep"]
-                    self.output.info("BUILD Dep %s VERSION %s" % (info.name, info.version))
+                    self.output.info("BUILD Dep %s VERSION %s" % (info.get_name('txt'), info.version))
```

Comment:
```
self.deps_cpp_info[“dep”].name
```
Is a documented thing. Seems this test should keep working without this change.

Reply:
That's true. 

Thinking about **Conan 2.0**: do you agree we should ban `.name`? or maybe it should return `ref.name` (we didn't want)? 

**Docs**: I think we should document `.get_name(<generator>)` instead of `name`.

Reply:
Yeah, agree:
- get_name will return build related/cpp_info information
- pkg information, as the package name, package version, and other package data, we will want to make it available, but maybe by other mean (node, pkg_info, or similar), not necessarily into the ``cpp_info`` data structure.
- From the user/consumer perspective there probably should be a parity between things defined in the ``self.cpp_info`` and things in ``self.deps_cpp_info["dep"]``.

So, we can deprecate name, remove it from the docs, and document .get_name() instead.